lhotse/cut.py

```diff
@@ -139,6 +139,17 @@ def supervisions_audio_mask(self) -> np.ndarray:
             mask[st:et] = 1
         return mask
 
+    def map_supervisions(self, transform_fn: Callable[[SupervisionSegment], SupervisionSegment]) -> 'CutUtilsMixin':
+        """
+        Modify the SupervisionSegments by `transform_fn` of this Cut.
+
+        :param transform_fn: a function that modifies a supervision as an argument.
+        :return: a modified Cut.
+        """
+        new_cut = fastcopy(self)
```

Comment:
This will be even faster:

    new_cut = fastcopy(self, supervisions=[s.map(transform_fn) for s in new_cut.supervisions])